### PR TITLE
demisto-sdk secrets: fix to ignore secrets by disable-secrets-detection

### PR DESCRIPTION
**Pull request by moishce**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/27356

## Description
fix to ignore secrets  by disable-secrets-detection

### CODE REVIEW

File: demisto_sdk/commands/secrets/tests/secrets_test.py [hunk 1]
```diff
@@ -313,3 +313,31 @@ def test_get_all_diff_text_files(self, mocker):
         validator.prev_ver = 'Testing_branch'
         assert validator.get_all_diff_text_files('master', True) == ['Packs/Integrations/integration/testing.py']
         assert validator.get_all_diff_text_files('master', False) == ['Packs/Integrations/integration/testing.py']
+
+    def test_remove_secrets_disabled_line(self):
+        """
+        Given
+            1. String with a line that are disable-secrets-detection
+            1. String with a lines whit "disable-secrets-detection-start" & "disable-secrets-detection-end"
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/secrets/tests/secrets_test.py:321]:
```suggestion
            1. String with a line containing "disable-secrets-detection" 
            1. String with a lines cpntaining "disable-secrets-detection-start" & "disable-secrets-detection-end"
```

File: demisto_sdk/commands/secrets/tests/secrets_test.py [hunk 2]
```diff
@@ -313,3 +313,31 @@ def test_get_all_diff_text_files(self, mocker):
...
+        When
+            Removing the lines from the file
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/secrets/tests/secrets_test.py:323]:
```suggestion
            Removing content that belongs to ignored lines
```

File: demisto_sdk/commands/secrets/tests/secrets_test.py [hunk 3]
```diff
@@ -313,3 +313,31 @@ def test_get_all_diff_text_files(self, mocker):
...
+        Then
+            Ensure secrets that that are in this lines isn't in the output.
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/secrets/tests/secrets_test.py:325]:
```suggestion
            Ensure secrets that are in these lines aren't in the output.
```

File: demisto_sdk/commands/secrets/tests/secrets_test.py [hunk 4]
```diff
@@ -313,3 +313,31 @@ def test_get_all_diff_text_files(self, mocker):
...
+        '''
+        file_contents = self.validator.remove_secrets_disabled_line(file_contents)
+        assert "8.8.8.8" not in file_contents
+
+        file_contents1 = '''
+        import
+        8.8.8.8 # disable-secrets-detection-start
+        4.4.4.4
+        end # disable-secrets-detection-end
+        '''
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/secrets/tests/secrets_test.py:340]: Is this the same behavior as was today? even if the comment is after the secret (like 8.8.8.8 it is being ignored?)

Comment (moishce) [reply demisto_sdk/commands/secrets/tests/secrets_test.py:340]: yes

Comment (ShahafBenYakir) [reply demisto_sdk/commands/secrets/tests/secrets_test.py:340]: Let's add another secret here and let's make sure it is picked up

**Comment by Itay4** [comment]
@moishce please update changelog

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,7 @@
 * Fixed an issue in the **validate** command where validation would fail due to a required version bump for packs which are not versioned.
 * Will use env var `DEMISTO_VERIFY_SSL` to determine if to use a secure connection for commands interacting with the Server when `--insecure` is not passed. If working with a local Server without a trusted certificate, you can set env var `DEMISTO_VERIFY_SSL=no` to avoid using `--insecure` on each command.
 * Unifier now adds a link to the integration documentation to the integration detailed description.
+* Fixed an issue where the **secrets** command don't ignore lines with the comment # disable-secrets-detection
```
Comment (ShahafBenYakir) [inline CHANGELOG.md:6]:
```suggestion
* Fixed an issue in the **secrets** command where ignore secrets were not skipped needed.
```

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**